lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -2089,4 +2072,33 @@ protected function switchPersisterContext($offset, $limit)
 
         $this->currentPersisterContext = $this->limitsHandlingContext;
     }
+
+    /**
+     * Retrieves class identifiers types
+     *
+     * @param ClassMetadata $class
+     * @return array
+     */
+    protected function getClassIdentifiersTypes(ClassMetadata $class)
+    {
+        $self = $this;
+
+        return array_map(function ($identifier) use ($class, $self) {
```

Comment:
The `$self` trick should no longer be needed, since we don't support older PHP versions anyway